### PR TITLE
Placeholder attribute is not output on frontend anymore.

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Fixes a bug where a `placeholder` attribute was output on the frontend for instance of the `RichText` block instruction.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Test this in WordPress SEO Premium.
* Create a post.
* Add a job posting block.
* In the location block, choose the 'office location' variation.
* Fill in the values of the office location.
* Save the post.
* View the post on the frontend.
* Check the source code of the post on the frontend.
* The source code for the location block should **not** have any `placeholder` attributes.
  ```html
  <div class="yoast-inner-container">
    <h3 data-id="Location">Location</h3>
    
    <h4 data-id="address-title">Address</h4>
    <p data-id="address">bla</p> <!-- <- These should not have any `placeholder` attributes. -->
    
    <h4 data-id="postal-code-title">Postal code</h4>
    <p data-id="postal-code">wjaoghagha</p>
    
    <h4 data-id="city-title">City</h4>
    <p data-id="city">aohawoghaoghaogh</p>
    
    <h4 data-id="region-title">Region</h4>
    <p data-id="region">agahowghaoghaogh</p>
    
    <h4 data-id="country-title">Country</h4>
    <p data-id="country">ohawogahwgoahgao</p>
  </div>
  ``` 

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: packages/schema-blocks/src/instructions/blocks/Heading.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class Heading extends RichTextBase {
 		class: string;
 		default: string;
 		placeholder: string;
-		keepPlaceholderOnFocus?: boolean;
+		keepPlaceholderOnFocus: boolean;
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/Heading.tsx:24]: why is this change necessary?

File: packages/schema-blocks/src/instructions/blocks/Heading.tsx [hunk 2]
```diff
@@ -117,8 +117,6 @@ export class Heading extends RichTextBase {
 			tagName: `h${ this.getHeadingLevel( props ) }` as keyof HTMLElementTagNameMap,
 			value: props.attributes[ this.options.name ] as string || this.options.value,
 			className: this.options.class,
-			placeholder: this.options.placeholder,
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/Heading.tsx:120]: did you intend to remove the placeholder?

File: packages/schema-blocks/src/instructions/blocks/RichText.ts [hunk 3]
```diff
@@ -29,8 +29,6 @@ export default class RichText extends RichTextBase {
 			tagName: this.options.tag,
 			value: props.attributes[ this.options.name ] as string,
 			className: this.options.class,
-			placeholder: this.options.placeholder,
```
Comment (increddibelly) [inline packages/schema-blocks/src/instructions/blocks/RichText.ts:32]: did you intentionally remove the placeholder?

File: packages/schema-blocks/src/instructions/blocks/Heading.tsx (hunk 1 above)
Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/Heading.tsx]: You are right. I saw that this attribute was either defined as optional (with the `?`) or not and decided that it should always be defined. 

But now I think it is better to keep it optional.

I made it optional.

File: packages/schema-blocks/src/instructions/blocks/Heading.tsx (hunk 2 above)
Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/Heading.tsx:120]: Yes, since it [is also added in the `edit` method of the `RichTextBase` instruction](https://github.com/Yoast/javascript/pull/1174/files#diff-0479fff95eef0e51fa1599f9a292610d7df2131ed01d407521afa650fd91690eL53-L55).

Since the `getBaseAttributes` instruction is used to get the base attributes for the component in the editor (in the block instruction's `edit` method), as well as on the frontend (in the block instruction's `save` method), this line meant that the `placeholder` attribute was added on the frontend as well.

Removing this line fixes this.

File: packages/schema-blocks/src/instructions/blocks/RichText.ts (hunk 3 above)
Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/RichText.ts:32]: Yes, see [the explanation in this comment](https://github.com/Yoast/javascript/pull/1174#discussion_r622138850).

**Review by andizer — APPROVED** [review]
CR Done

**Comment by andizer** [comment]
Acceptance 👍
